backend/azure/keyvault.go

```diff
@@ -29,11 +31,21 @@ type keyvaultClient interface {
 
 // getKeyvaultClient is a variable that holds the function to create a new keyvaultClient
 // it will be overwritten in tests
-var getKeyvaultClient = func(keyVaultURL string) (keyvaultClient, error) {
-	cred, err := azidentity.NewDefaultAzureCredential(nil)
-	if err != nil {
-		return nil, fmt.Errorf("getting default credentials: %s", err)
+var getKeyvaultClient = func(keyVaultURL, clientID string) (keyvaultClient, error) {
+	var err error
+	var cred azcore.TokenCredential
+	if clientID == "" {
+		cred, err = azidentity.NewDefaultAzureCredential(nil)
+	} else {
+		opts := azidentity.ManagedIdentityCredentialOptions{ID: azidentity.ClientID(clientID)}
+		cred, err = azidentity.NewManagedIdentityCredential(&opts)
 	}
+	if err != nil && clientID == "" {
+		return nil, fmt.Errorf("clientID not provided, could not get credentials: %s", err)
+	} else if err != nil {
+		return nil, fmt.Errorf("getting identity credentials: %s", err)
+	}
```

Comment:
Should we keep the error handling within each if-case ?

```suggestion
	if clientID == "" {
		cred, err = azidentity.NewDefaultAzureCredential(nil)
		if err != nil {
			return nil, fmt.Errorf("clientID not provided, could not get credentials: %s", err)
		}
	} else {
		opts := azidentity.ManagedIdentityCredentialOptions{ID: azidentity.ClientID(clientID)}
		cred, err = azidentity.NewManagedIdentityCredential(&opts)
		if err != nil {
			return nil, fmt.Errorf("getting identity credentials: %s", err)
		}
	}
```